lib/src/rewrite.rs

```diff
@@ -553,16 +565,23 @@ impl<'settings, 'repo> DescendantRebaser<'settings, 'repo> {
             return Ok(());
         }
 
-        // Don't create commit where one parent is an ancestor of another.
-        let head_set: HashSet<_> = self
-            .mut_repo
-            .index()
-            .heads(&mut new_parent_ids.iter())
-            .into_iter()
-            .collect();
+        // If specified, don't create commit where one parent is an ancestor of another.
+        let new_parent_ids: Vec<_> = if self.options.no_ancestor_merge {
+            let head_set: HashSet<_> = self
+                .mut_repo
+                .index()
+                .heads(&mut new_parent_ids.iter())
+                .into_iter()
+                .collect();
+            new_parent_ids
+                .into_iter()
+                .filter(|new_parent| head_set.contains(new_parent))
```

Comment:
nit: (at your option) `.retain()` could be used, but that would make the vec mutable.

Reply:
Done.